js/src/api/contract/contract.js

```diff
@@ -269,6 +217,11 @@ export default class Contract {
       }
     }
 
+    return this._subscribe(event, options, callback);
+  }
+
+  _subscribe (event = null, _options, callback) {
+    const subscriptionId = this._subscriptions.length;
```

Comment:
Implemented as per your idea. (Mine is not quite feasible since filterId only gets returned after the promise succeeds, so it is not available for the sync return value of the id to the caller.) 

Could make it return a promise that returns the subscriptionId, but will keep that idea on ice for now since the API interfaces needs to be consistent and don't want to expand the PR into non-related areas (we also have subscriptions for things like eth_blockNumber). Will need to think it through, making it Promis-y for the subscriptionId does fit in with the overall API approach where we use promises for most calls.